gips/data/gpm/gpm.py

```diff
@@ -0,0 +1,335 @@
+
+from __future__ import print_function
+
+import os
+import sys
+import re
+import datetime
+
+import urllib
+import urllib2
+
+import math
+import numpy as np
+import requests
+import ftplib
+import gippy
+from gippy.algorithms import Indices
+from gips.data.core import Repository, Asset, Data
+from gips.utils import VerboseOut, settings
+from gips import utils
+
+
+class gpmRepository(Repository):
+    name = 'GPM'
+    description = 'Global Precipitation Measurement Mission (GPM)'
+    # NASA assets require special authentication
+    _host = "arthurhou.pps.eosdis.nasa.gov"
+    _tile_attribute = 'tileid'
+
+class gpmAsset(Asset):
+    Repository = gpmRepository
+    _sensors = {'GPM': {'description': 'Integrated Multi-satellite Retrievals for GPM'}}
+    _assets = {
+        'IMERG-DAY-FINAL': {
+                        'host': "arthurhou.pps.eosdis.nasa.gov",
+            'pattern': r'3B-DAY-GIS\.MS\.MRG\.3IMERG\..{8}-S.{6}-E.{6}\..{4}\..{4}\.tif',
+            'description': 'Daily Precipitation Rate (mm/hr) at 0.1 degrees - Research Run using the IMERG Algorithm'
+                           'developed for the GPM constellation',
+            'path': '/gpmdata/',
+            'startdate': datetime.date(2014, 3, 12),
+            'latency': 1,
+        },
+        'IMERG-DAY-LATE': {
+            'host': "jsimpson.pps.eosdis.nasa.gov",
+            'pattern': r'3B-HHR-L\.MS\.MRG\.3IMERG\..{8}-S.{6}-E.{6}\..{4}\..{4}\.1day.tif',
+            'description': 'Daily Accumulated Precipitation (in mm) at 23:59 UTC at 0.1 degrees - Production Run '
+                           'using the IMERG Algorithm developed for the GPM constellation',
+            'path': '/NRTPUB/imerg/gis/',
+            'startdate': datetime.date(2014, 3, 12),
+            'latency': 1,
+        },
+        'IMERG-DAY-EARLY': {
+            'host': "jsimpson.pps.eosdis.nasa.gov",
+            'pattern': r'3B-HHR-L\.MS\.MRG\.3IMERG\..{8}-S.{6}-E.{6}\..{4}\..{4}\.1day.tif',
+            'description': 'Daily Accumulated Precipitation (in mm) at 06:59 UTC 0.1 degrees - Production Run '
+                           'using the IMERG Algorithm developed for the GPM constellation',
+            'path': '/NRTPUB/imerg/gis/',
+            'startdate': datetime.date(2014, 3, 12),
+            'latency': 1,
+        },
+        'IMERG-DAY-MID': {
+            'host': "jsimpson.pps.eosdis.nasa.gov",
+            'pattern': r'3B-HHR-L\.MS\.MRG\.3IMERG\..{8}-S.{6}-E.{6}\..{4}\..{4}\.1day.tif',
+            'description': 'Daily Accumulated Precipitation (in mm) at 14:59 0.1 degrees - Production Run '
+                           'using the IMERG Algorithm developed for the GPM constellation',
+            'path': '/NRTPUB/imerg/gis/',
+            'startdate': datetime.date(2014, 3, 12),
+            'latency': 1,
+        },
+        '3B42-DAY-LATE': {
+            'host': "trmmopen.pps.eosdis.nasa.gov",
+            'pattern': r'3B42RT\..{10}[\.7]*\.1day.tif',
```

Comment:
Changed to r'^3B42RT\..{10}\.7\.1day.tif$'

Reply:
Hi @subitc -- did you push this change somewhere that I'm not seeing?  I ask because we have someone here who is test driving the driver, and ran into a problem that roots in this missing `$`.

Reply:
Pushed just now. Sorry, I had gone through like half of the changes requested.

Reply:
No problem.  I just remembered seeing your comment "changed" come through, but didn't see the code change.